--- gmp-5.0.5/configure.in.x32abi~	2012-05-06 13:19:49.000000000 +0200
+++ gmp-5.0.5/configure.in	2012-07-20 12:53:14.663628004 +0200
@@ -1614,6 +1614,21 @@ case $host in
 	    gcc_64_cflags="$gcc_64_cflags -std=gnu99"
 	    ;;
 	esac
+
+	# X32 support.
+	case x"$path_64" in
+	  xx86_64*)
+	    abilist="x32 64 32"
+	    path_x32="$path_64"
+	    limb_x32=longlong
+	    cclist_x32="gcc"
+	    gcc_x32_cflags="-O2 -mx32"
+	    any_x32_testlist="sizeof-long-4"
+	    CALLING_CONVENTIONS_OBJS_x32='amd64call.lo amd64check$U.lo'
+	    SPEED_CYCLECOUNTER_OBJ_x32=x86_64.lo
+	    cyclecounter_size_x32=2
+	    ;;
+	esac
 	;;
     esac
     ;;
@@ -3244,7 +3259,7 @@ if test "$gmp_asm_syntax_testing" != no;
 	      GMP_INCLUDE_MPN(x86/darwin.m4) ;;
 	  esac
           ;;
-        64)
+        64|x32)
           GMP_INCLUDE_MPN(x86_64/x86_64-defs.m4)
           AC_DEFINE(HAVE_HOST_CPU_FAMILY_x86_64)
 	  case $host in
